packages/arb-token-bridge-ui/src/util/deposits/fetchDeposits.ts

```diff
@@ -1,136 +0,0 @@
-export const fetchDeposits = async ({
-  sender,
-  receiver,
-  fromBlock,
-  toBlock,
-  l1Provider,
-  l2Provider,
-  pageSize = 10,
-  pageNumber = 0,
-  searchString = ''
-}: FetchDepositParams): Promise<Transaction[]> => {
-  if (typeof sender === 'undefined' && typeof receiver === 'undefined')
-    return []
-  if (!l1Provider || !l2Provider) return []
-
-  const l1ChainId = (await l1Provider.getNetwork()).chainId
-  const l2ChainId = (await l2Provider.getNetwork()).chainId
-
-  const nativeCurrency = await fetchNativeCurrency({ provider: l2Provider })
-
-  if (!fromBlock) {
-    fromBlock = 0
-  }
-
-  if (!toBlock) {
-    // if toBlock hasn't been provided by the user
-
-    // fetch the latest L1 block number thorough subgraph first
-    let latestL1BlockNumber = await tryFetchLatestSubgraphBlockNumber(
-      'L1',
-      l2ChainId
-    )
-
-    // if the previous call returns 0, then fetch the latest block on-chain
-    if (!latestL1BlockNumber) {
-      latestL1BlockNumber = await l1Provider.getBlockNumber()
-    }
-
-    toBlock = latestL1BlockNumber
-  }
-
-  const depositsFromSubgraph = await fetchDepositsFromSubgraph({
-    sender,
-    receiver,
-    fromBlock,
-    toBlock,
-    l2ChainId,
-    pageSize,
-    pageNumber,
-    searchString
-  })
-
-  const ethDepositsFromSubgraph: Transaction[] = depositsFromSubgraph.map(
-    (tx: FetchDepositsFromSubgraphResult) => {
-      const isEthDeposit = tx.type === 'EthDeposit'
-
-      const assetDetails = {
-        assetName: nativeCurrency.symbol,
-        assetType: AssetType.ETH,
-        tokenAddress: ''
-      }
-
-      if (!isEthDeposit) {
-        // update some values for token deposit
-        const symbol = tx.l1Token?.symbol || ''
-
-        assetDetails.assetName = symbol
-        assetDetails.assetType = AssetType.ERC20
-        assetDetails.tokenAddress = tx?.l1Token?.id || ''
-      }
-
-      const amount = isEthDeposit ? tx.ethValue : tx.tokenAmount
-
-      const tokenDecimals = tx?.l1Token?.decimals ?? defaultErc20Decimals
-      const decimals = isEthDeposit ? nativeCurrency.decimals : tokenDecimals
-
-      return {
-        type: 'deposit-l1',
-        status: 'pending',
-        value: utils.formatUnits(amount || 0, decimals),
-        txID: tx.transactionHash,
-        tokenAddress: assetDetails.tokenAddress,
-        sender: tx.sender,
-        destination: tx.receiver,
-
-        assetName: assetDetails.assetName,
-        assetType: assetDetails.assetType,
-
-        l1NetworkID: String(l1ChainId),
-        l2NetworkID: String(l2ChainId),
-        blockNumber: Number(tx.blockCreatedAt),
-        timestampCreated: tx.timestamp,
-        isClassic: tx.isClassic,
-
-        chainId: l2ChainId,
-        parentChainId: l1ChainId
-      }
-    }
-  )
-
-  const finalTransactions: Transaction[] = await Promise.all(
-    ethDepositsFromSubgraph.map(depositTx =>
-      updateAdditionalDepositData(depositTx, l1Provider, l2Provider)
-    )
-  )
```

Comment:
Let's move this out of `fetchDeposits` and into `useDeposits` in a separate PR. That way we could reuse the current `fetchDeposits` for the new transaction history and reduce the diff. Same thing for withdrawals.

Reply:
https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1311